### PR TITLE
Adapt to work with arrays, and to accept values that aren't objects.

### PR DESCRIPTION
**Pull request by mrmurphy**
This change accommodates arrays, objects that have arrays, and also values that aren't objects or arrays.

I've added some tests that explain better than my words do :)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
-};
\ No newline at end of file
+    var o = {};
+    forOwn(obj, function (value, key) {
+      o[key] = omitDeep(value, props);
+    });
+    return omit(o, props);
+  }
+
+  if (input === undefined) {
```
Comment (jonschlinkert) [inline index.js]: I think this should be `typeof input === 'undefined'`

File: index.js [hunk 2]
```diff
@@ -1,19 +1,35 @@
...
 
-  forOwn(obj, function (value, key) {
-    if (isObject(value)) {
-      o[key] = omitDeep(value, props);
-    } else {
-      o[key] = value;
+module.exports = function omitDeep(input, props) {
+  function omitDeepOnOwnProps(obj) {
+    if (!isArray(obj) && !isObject(obj)) {
+      return obj
```
Comment (jonschlinkert) [inline index.js]: you missed a few semi-colons :)

**Comment by jonschlinkert** [comment]
besides the comments, looks good. I'll merge as soon as we get those updated. thanks!

File: index.js (hunk 2 above)
Comment (mrmurphy) [inline index.js]: Whoops! Sorry about that. That's the convention we have at work, and I wasn't paying attention :)

File: index.js (hunk 1 above)
Comment (mrmurphy) [inline index.js]: Will do!

**Comment by mrmurphy** [comment]
There we go, thanks! And that's a great profile pic :)

**Comment by jonschlinkert** [comment]
thanks! and that's a great profile pic you have there yourself! lol. yeah nothing comes close to being as awesome as having kids :)

**Comment by pedrombafonso** [comment]
+1

**Comment by YagoQuinoy** [comment]
+1

**Comment by andrewmclagan** [comment]
Haha...You two look like brothers. Two things I would mention:

1. This is not documented
2. Does not work when omitting nesting within an array e.g:

```javascript
omit({
  foo: [{ bar: [{ bam: "1", safe: true }] }],
}, ['foo.bar.bam']);
```

**Pull request merged**